### PR TITLE
Refactor CLI config to allow for non-singleton config

### PR DESCRIPTION
**Pull request by csansoon**
The VSCode extension should use functions and libraries defined in the CLI package to configure, run, manage files and projects.

The current implementation of CLI is using a `config` singleton that that is initiated from the terminal cwd. The expected use for VSCode, instead, could have different files from different Latitude projects opened in the same Editor, and it should handle them with their own config accordingly, which does not have to be from the workspace source either.

To fix this, now some CLI functions may accept a custom `CLIConfig` class, which will use the singleton by default.

### CODE REVIEW

File: packages/cli/rollup.config.mjs [hunk 1]
```diff
@@ -23,11 +23,15 @@ const packageVersion =
...
 export default {
-  input: 'src/index.ts',
+  input: {
+    index: 'src/index.ts',
+    cli: 'src/cli.ts',
+  },
   cache: false, // This is generating stale builds in watch mode
   output: {
-    file: 'dist/index.js',
+    dir: 'dist',
```
Comment (andresgutgon) [inline packages/cli/rollup.config.mjs:32]: Check `dist` folder looks good, that types are exported correctly

File: packages/cli/src/cli.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  .option('--watch', 'Re-run the query each time the query file changes')
+  .option(
+    '--param',
+    'Add a parameter to the query. Use the format --param <name>=<value>',
+  )
+  .example('run --watch users')
+  .example('run users --param user_id=foo')
+  .action(command('runCommand'))
+
+function cliFactory() {
```
Comment (andresgutgon) [inline packages/cli/src/cli.ts:55]: Put this on top. I always "flipo" went I start reading a file and don't see the import or the function

Comment (csansoon) [reply packages/cli/src/cli.ts:55]: Yeah I was wondering why is this even defined in a function

File: packages/cli/src/lib/maybeSetupApp/index.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
-import config from '$src/config'
+import { CLIConfig } from '$src/config'
 import fsExtra from 'fs-extra'
 import path from 'path'
 import setupApp from '../setupApp'
 import { LATITUDE_FOLDER } from '$src/commands/constants'
 
-export default async function maybeSetupApp() {
-  const hasApp = fsExtra.existsSync(path.join(config.cwd, LATITUDE_FOLDER))
+export default async function maybeSetupApp(config: CLIConfig = CLIConfig.getInstance()) {
```
Comment (andresgutgon) [inline packages/cli/src/lib/maybeSetupApp/index.ts:7]: Ok so what this is doing is getting the singleton as default and you're setting `config` to pass from the VSCode extension?

File: packages/cli/src/cli.ts (hunk 2 above)
Comment (andresgutgon) [reply packages/cli/src/cli.ts:55]: No need right?

File: packages/cli/rollup.config.mjs (hunk 1 above)
Comment (csansoon) [reply packages/cli/rollup.config.mjs:32]: Hmm, good catch! Now `src/index.ts` and `src/cli.ts` are indeed being compiled into `dist/index.js` and `dist/cli.js` respectively, but the type files are being exported to `dist/src/*.d.ts`.

Would changing the package.json to `types: "dist/src/index.d.ts` work? Or do I have to change the tsconfig.json "rootDir"?

File: packages/cli/src/lib/maybeSetupApp/index.ts (hunk 3 above)
Comment (csansoon) [reply packages/cli/src/lib/maybeSetupApp/index.ts:7]: The VSCode extension will create a custom `config` depending on the open Latitude file, with its own `source` directory. Currently I am not using `maybeSetupApp`, but it just defaults to the config singleton.

File: packages/cli/src/cli.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  .option(
+    '--param',
+    'Add a parameter to the query. Use the format --param <name>=<value>',
+  )
+  .example('run --watch users')
+  .example('run users --param user_id=foo')
+  .action(command('runCommand'))
+
+
+async function init() {
```
Comment (andresgutgon) [inline packages/cli/src/cli.ts:67]: 👾

**Review by andresgutgon — DISMISSED** [review]
Do QA, run all commands with all the combinations

**Review by andresgutgon — APPROVED** [review]

**Pull request merged**